File: 2860/main.go
package main

import (
	"fmt"
	"sort"
)

func main() {
	fmt.Println(countWays([]int{1,1})) // 2
}
func countWays(nums []int) int {
	n := len(nums)
	res := 0 
	sort.Ints(nums)
	for k := 0; k <= n; k++ {
		if k > 0 && nums[k-1] >= k {
			continue
		}
		if k < n && nums[k] <= k {
			continue
		}
		res++
	}
	return res
}

File: 2860/main.rs
struct Solution {}

impl Solution {
    pub fn count_ways(mut nums: Vec<i32>) -> i32 {
        let n = nums.len();
        let mut res = 0; 
        nums.sort();
        for k in 0..n+1 {
            if k > 0 && nums[k-1] >= k as i32 {
                continue
            }
            if k < n && nums[k] <= k as i32 {
                continue
            }
            res += 1;
        }
        return res
    }
}

fn main() {
    println!("{}", Solution::count_ways(vec![1, 1])) // 2
}